registration-processor/registration-processor-info-storage-service/src/test/java/io/mosip/registration/processor/packet/storage/utils/ABISHandlerUtilTest.java

```diff
@@ -1,164 +1,170 @@
+		matchedRids.add("10002100820001420210108085102");
+		List<AbisResponseDto> abisResponseDtoList = new ArrayList<>();
+		matchedRids.forEach(matchedRid -> {
+			AbisResponseDto abisResponseDto = new AbisResponseDto();
+			abisResponseDto.setId(matchedRid);
+			abisResponseDtoList.add(abisResponseDto);
+		});
+
+		when(packetInfoManager.getAbisResponseRecords(regBioRefIds.get(0),
+				latestTransactionId, AbisConstant.IDENTIFY)).thenReturn(abisResponseDtoList);
+
+		List<AbisResponseDetDto> abisResponseDetDtoList = new ArrayList<>();
+
+		matchedRids.forEach(matchedRid -> {
+			AbisResponseDetDto abisResponseDto = new AbisResponseDetDto();
+			abisResponseDto.setMatchedBioRefId(matchedRid);
+			abisResponseDetDtoList.add(abisResponseDto);
+		});
+		for (AbisResponseDetDto dto : abisResponseDetDtoList) {
+			AbisResponseDetDto responseDetDto = new AbisResponseDetDto();
+			responseDetDto.setMatchedBioRefId(dto.getMatchedBioRefId());
+			when(packetInfoManager.getAbisResponseDetails(dto.getMatchedBioRefId())).thenReturn(Lists.newArrayList(responseDetDto));
+		}
+
+		when(packetInfoDao.getAbisRefRegIdsByMatchedRefIds(matchedRids)).thenReturn(matchedRids);
+
+		when(packetInfoDao.getWithoutStatusCode(matchedRids, RegistrationStatusCode.REJECTED.toString()))
+				.thenReturn(registrationStatusEntityList);
+	
+
+		when(idRepoService.getUinByRid("10002100820001420210108085103", new String())).thenReturn("123456789");
+		when(idRepoService.getUinByRid("10002100820001420210108085102", new String())).thenReturn("987654321");
+
+	}
+
+	@Test
+	public void testProcesssedWithUniqueUin() throws ApisResourceAccessException, JsonProcessingException, PacketManagerException, IOException, io.mosip.kernel.core.exception.IOException {
+
+		List<String> uniqueRids = abisHandlerUtil.getUniqueRegIds(registrationId, registrationType, ProviderStageName.BIO_DEDUPE);
+
+		assertEquals(3, uniqueRids.size());
+	}
+
+	@Test
+	public void testProcesssedWithSameUin() throws ApisResourceAccessException, JsonProcessingException, PacketManagerException, IOException, io.mosip.kernel.core.exception.IOException {
+
+		when(idRepoService.getUinByRid(anyString(), anyString())).thenReturn("987654321");
+
+		List<String> uniqueRids = abisHandlerUtil.getUniqueRegIds(registrationId, registrationType, ProviderStageName.BIO_DEDUPE);
+		// expected to pick only 1 rid from processedMatchedIds list because same uin. Total should be 1(inprogress) + 1(processed)
+		assertEquals(2, uniqueRids.size());
+	}
+
+	@Test
+	public void testDonotReturnRejected() throws ApisResourceAccessException, JsonProcessingException, PacketManagerException, IOException, io.mosip.kernel.core.exception.IOException {
+
+		List<String> uniqueRids = abisHandlerUtil.getUniqueRegIds(registrationId, registrationType, ProviderStageName.BIO_DEDUPE);
+		// expected to pick only processingandprocessed list i.e 3 records.
+		assertNotEquals(matchedRids.size(), uniqueRids.size());

```

Comment:
Can we make this assert with exact number? so the expectation matches the comment?

Reply:
fixed